src/zodbbrowser/browser.py

```diff
@@ -85,10 +107,47 @@ def jar(self):
                 raise Exception("ZODB connection not available for this request")
             return obj._p_jar
 
-    @property
+    @Lazy
     def readonly(self):
         return self.jar.isReadOnly()
 
+    def findObjectFromOID(self, oid, tid=None):
+        p_oid = p64(oid)
+        try:
+            return self.jar.get(p_oid)
+        except POSKeyError as error:
+            if self.references is not None:
+                for referred in self.references.getBackwardReferences(p_oid):
+                    try:
+                        # This will access the original object and
+                        # create a ghost.
+                        ZodbObjectState(self.jar.get(p64(referred)), tid)
+                    except POSKeyError:
+                        continue
+                    # We loaded a referrer. Maybe we can try to load
+                    # the object again.
+                    try:
+                        return self.jar.get(p_oid)
+                    except POSKeyError:
+                        pass
```

Comment:
This for loop could use a comment explaining what it is that it's trying to do and why.